src/main/java/org/ice4j/stack/StunClientTransaction.java

```diff
@@ -486,27 +301,21 @@ void cancel()
      */
     public void handleResponse(StunMessageEvent evt)
     {
-        lock.lock();
-        try
-        {
-            TransactionID transactionID = getTransactionID();
-
-            logger.log(Level.FINE, "handleResponse tid " + transactionID);
-            if(!Boolean.getBoolean(StackProperties.KEEP_CRANS_AFTER_A_RESPONSE))
-                cancel();
-
-            responseCollector.processResponse(
-                    new StunResponseEvent(
-                            stackCallback,
-                            evt.getRawMessage(),
-                            (Response) evt.getMessage(),
-                            request,
-                            transactionID));
-        }
-        finally
+        TransactionID transactionID = getTransactionID();
+
+        logger.log(Level.FINE, "handleResponse tid " + transactionID);
```

Comment:
maybe change this to a direct call since you're tweaking it (`logger.fine` i think?)